iFixit/react-commerce
# Define _paq on window if it isn't yet defined: fixes setEcommerceView issue

...and add `_paq` to the window interface. (Not sure if the latter is necessary, but I was getting messages from my editor about the `window` object not having the property or something like that.)

Do this inside an exported function that encapsulates the `_paq = []` definition and `_paq.push` functionality.

> `Property '_paq' does not exist on type 'Window & typeof globalThis'.`

The `setEcommerceView` functionality didn't seem to be working properly before because `_paq` wasn't being created on the global `window` object if `_paq` wasn't yet defined on it.

https://github.com/iFixit/react-commerce/blob/matomo--fix-set-ecommerce-view/packages/analytics/matomo/index.tsx#L34-L37

# QA notes
At the beginning of viewing a web page, two requests go out to Matomo:

- to fetch the tracking JS code
- to track a page view

The page view tracking request on **product pages currently do not include the SKU, (full) product name, and the price** (converted to USD). This should amend that.

Closes https://github.com/iFixit/react-commerce/issues/1415
